src/Microsoft.AspNetCore.Sockets.Common/Internal/ChannelConnection.cs

```diff
@@ -7,21 +7,33 @@ namespace Microsoft.AspNetCore.Sockets.Internal
 {
     public static class ChannelConnection
     {
+        public static ChannelConnection<TIn, TOut> Create<TIn, TOut>(Channel<TIn> input, Channel<TOut> output)
+        {
+            return new ChannelConnection<TIn, TOut>(input, output);
+        }
+
         public static ChannelConnection<T> Create<T>(Channel<T> input, Channel<T> output)
```

Comment:
I tried removing this overload but it did not really work because then I could not assign it to `IChannelConnection<T>` without cast.